### PR TITLE
Add basic circleci config

### PR DESCRIPTION
**Pull request by rtburns-jpl**
Currently builds python package and runs download script. Downloaded data is not yet used.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          name: "Install build system dependencies"
+          command: |
+            sudo apt-get update
+            sudo apt-get install python3 python3-setuptools python3-pip wget
+            python3 -m pip install --user build
+      - checkout
+      - run:
+          name: "Build s1-reader python package"
+          command: |
+            python3 -m build --sdist --wheel .
```
Comment (hfattahi) [inline .circleci/config.yml:19]: Thanks @rtburns-jpl. s1-reader depends on isce3 . The rest of dependencies are [here](https://github.com/opera-adt/s1-reader/blob/main/requirements.txt). As we discussed we want to move forward with isce3 on conda package.

**Comment by vbrancat** [comment]
@rtburns-jpl any progress on this? :)

**Comment by rtburns-jpl** [comment]
Yeah just pushed what I have locally, let's see what the CI thinks

**Comment by rtburns-jpl** [comment]
Cool, CI is happy. Will just push some small fixes for the codacy linter.

**Comment by rtburns-jpl** [comment]
Ok all green now. This is ready for review :)

**Review by LiangJYu — APPROVED** [review]
LGTM. I have a question that will not hold up the PR.

**Review by vbrancat — APPROVED** [review]
Thanks, @rtburns-jpl ! LGTM.

**Comment by vbrancat** [comment]
@rtburns-jpl ready to merge? :D

**Comment by hfattahi** [comment]
@rtburns-jpl anything holding this PR?

**Comment by rtburns-jpl** [comment]
Oh nothing at all, let's merge it!

**Pull request merged**